lib/dry/validation/failures.rb

```diff
@@ -60,6 +60,11 @@ def failure(message, tokens = EMPTY_HASH)
         opts << { message: message, tokens: tokens, path: path }
         self
       end
+
+      # @api private
+      def empty?
+        opts.any?
```

Comment:
this should be `opts.empty?` and in `rule_error?` it should do `!key(path).empty?`

Reply:
Oh, damn it! I think yesterday I had my brain upside down... Now it's ok :sweat_smile: